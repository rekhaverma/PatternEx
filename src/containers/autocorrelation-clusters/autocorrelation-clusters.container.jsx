import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import moment from 'moment';

import { selectCluster, sortClusterBy } from 'model/selectors';
import { updateClusterSortBy } from 'model/actions';
import ClustersToolbar from 'components/clusters-toolbar';

const options = [
  { 'id': 'severity', 'content': 'Severity' },
  { 'id': 'newest', 'content': 'Newest' },
  { 'id': 'oldest', 'content': 'Oldest' },
];

/**
 *  Function to filter the clusters based on the selected filters
 * (timeline filter/ heatmap filter) on dashboard
 * @param {object} state
 * @param {array} items
 */
const filterList = (state, items) => {
  const filters = state.raw.getIn(['relations', 'filters']).toJS();
  const highlighted = state.app.maliciousActivity.toJS().highlighted;
  let filteredData = items;
  if (Object.keys(filters.active).length > 0) {
    filteredData = items
      .filter(el => el.seed_type === Object.keys(filters.active)[0])
      .filter(el => el.tag_id === Object.values(filters.active)[0]);
  }

  if (highlighted !== '') {
    filteredData = filteredData
      .filter((el) => {
        const utcDate = moment.utc(el.ts).format('MM - DD - YYYY');
        return highlighted.format('MM - DD - YYYY') === utcDate;
      });
  }

  return filteredData;
};

export const AutocorrelationClusters = props => (
  <ClustersToolbar {...props} options={options} />
);
AutocorrelationClusters.propTypes = {
  'list': PropTypes.array.isRequired,
  'sortClusterBy': PropTypes.string.isRequired,
  'onClusterClick': PropTypes.func.isRequired,
  'updateClusterSortBy': PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  'list': filterList(state, selectCluster(state)),
  'sortClusterBy': sortClusterBy(state),
});

const mapDispatchToProps = dispatch => ({
  'updateClusterSortBy': (...args) => dispatch(updateClusterSortBy(...args)),
});

export default connect(mapStateToProps, mapDispatchToProps)(AutocorrelationClusters);
