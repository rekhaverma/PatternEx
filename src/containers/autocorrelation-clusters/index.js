export { default } from './autocorrelation-clusters.container';
