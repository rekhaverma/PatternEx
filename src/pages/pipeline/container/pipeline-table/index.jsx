export { default } from './pipeline-table.container';
