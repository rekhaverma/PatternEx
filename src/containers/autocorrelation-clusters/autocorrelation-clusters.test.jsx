import React from 'react';

import { AutocorrelationClusters } from './autocorrelation-clusters.container';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<AutocorrelationClusters />', () => {
  describe('render: ', () => {
    it('should match with snapshot', () => {
      const props = {
        list: [],
        sortClusterBy: '',
        onClusterClick: jest.fn(),
        updateClusterSortBy: jest.fn(),
      };
      const wrapper = shallow(<AutocorrelationClusters {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
