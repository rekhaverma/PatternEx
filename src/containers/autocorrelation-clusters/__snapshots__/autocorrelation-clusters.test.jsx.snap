// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutocorrelationClusters /> render:  should match with snapshot 1`] = `
<Connect(ClustersToolbar)
  list={Array []}
  onClusterClick={[Function]}
  options={
    Array [
      Object {
        "content": "Severity",
        "id": "severity",
      },
      Object {
        "content": "Newest",
        "id": "newest",
      },
      Object {
        "content": "Oldest",
        "id": "oldest",
      },
    ]
  }
  sortClusterBy=""
  updateClusterSortBy={[Function]}
/>
`;
